src/llscan.cc

```diff
@@ -1242,12 +1243,20 @@ void LLScan::ScanMemoryRanges(FindJSObjectsVisitor& v) {
       for (size_t j = 0; j + addr_size <= loaded;) {
         uint64_t value;
 
-        if (addr_size == 4)
+        if (addr_size == 4) {
           value = *reinterpret_cast<uint32_t*>(&block[j]);
-        else if (addr_size == 8)
+          if (process_byte_order != HOST_BYTE_ORDER) {
+            value = __builtin_bswap32(value);
+          }
+        }
+        else if (addr_size == 8) {
           value = *reinterpret_cast<uint64_t*>(&block[j]);
-        else
+          if (process_byte_order != HOST_BYTE_ORDER) {
+            value = __builtin_bswap64(value);
+          }
+        } else {
```

Comment:
gcc and clang lower it to `__builtin_memcpy()` which emits simple loads and stores for word-sized copies (on architectures where alignment isn't an issue.)  I'd be surprised if it was in any way slower.